### PR TITLE
Masked normalization infrastructure and strided masked softmax

### PR DESCRIPTION
**Pull request by pearu**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68584
* #68463
* #68461
* __->__ #68333

Differential Revision: [D32564435](https://our.internmc.facebook.com/intern/diff/D32564435)

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -5521,6 +5521,22 @@ def sample_inputs_softmax_variant(op_info, device, dtype, requires_grad, with_dt
     ]
 
 
+def sample_inputs_masked_softmax(op_info, device, dtype, requires_grad, with_dtype=False, **kwargs):
+    """Sample inputs for masked softmax and log_softmax.
+
+    Masked reduction operator is a reduction operator with trailing
```
Comment (cpuhrsch) [inline torch/testing/_internal/common_methods_invocations.py:5527]: nit: do you mean "masked normalization"?

**Review by cpuhrsch — APPROVED** [review]
LGTM.

**Comment by cpuhrsch** [comment]
@ngimel - you're probably exited to see this, maybe you can take a quick look as well?

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (pearu) [reply torch/testing/_internal/common_methods_invocations.py]: yes, fixed.

**Comment by cpuhrsch** [comment]
@pearu - could you rebase the stack please?

**Comment by cpuhrsch** [comment]
@cpuhrsch has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32564435).

**Review by cpuhrsch — APPROVED** [review]
